catalog/task/slack-webhook-notification/0.1/slack-webhook-notification.yaml

```diff
@@ -0,0 +1,68 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "release"
+  name: slack-webhook-notification
+spec:
+  description: >-
+    Sends message to slack using incoming webhook
+  params:
+    - name: message
+      description: Message to be sent
+    - name: extraDataJsonPath
+      description: path to extraData json file
+      default: "$(workspaces.data.path)/extra_data.json"
+    - name: secretName
+      description: |
+        Secret with at least one key where value is webhook URL for slack.
+        eg. oc create secret generic my-secret --from-literal team1=https://hooks.slack.com/services/XXX/XXXXXX 
+        --from-literal team2=https://hooks.slack.com/services/YYY/YYYYYY
+      default: slack-webhook-notification-secret
+  volumes:
+    - name: webhook-secret
+      secret:
+        secretName: $(params.secretName)
+        optional: true
+  steps:
+    - name: send-message
+      image: >-
+        quay.io/hacbs-release/release-utils@sha256:5733ece907aa70d6ebced36484c936731e8b27bfcf87fed226a0ecf95047a6b8
+      volumeMounts:
+        - name: webhook-secret
+          mountPath: "/etc/secrets"
+          readOnly: true
+      env:
+        - name: HOME
+          value: /tekton/home
+        - name: MESSAGE
+          value: $(params.message)
+      script: |
+        #!/usr/bin/env bash
+        set -x
+
+        # get keyName from ExtraData
+        echo "extraData: $(params.extraDataJsonPath)"
+        cat "$(params.extraDataJsonPath)"
+        keyName=$(cat "$(params.extraDataJsonPath)" | jq -r '."slack-webhook-notification-secret-keyname" // empty')
+
+        if [ "${keyName}" == "" ] ; then
+          echo "No secret key name provided via 'slack-webhook-notification-secret-keyname' key in extraData"
+          exit 0
```

Comment:
so we don't want to fail here?

Reply:
I will update the `echo`. But essentially having a `keyName` is what enables the task. Therefore `no keyName` = `<disabled>`

Reply:
Yeah, I just wanted to ask because sometimes when we call a task and it is missing the data to perform its main function, we fail, and others we don't. It seems like this is one we don't want to fail on